core/resources/material.glsl

```diff
@@ -0,0 +1,21 @@
+struct Material {
+	#ifdef MATERIAL_EMISSION
+	vec4 emission;
+	#endif
+
+	#ifdef MATERIAL_AMBIENT
+	vec4 ambient;
+	#endif 
+
+	#ifdef MATERIAL_DIFFUSE
+	vec4 diffuse;
+	#endif
+
+	#ifdef MATERIAL_SPECULAR
+	vec4 specular;
+	float shininess;
+	#endif
+};
+
+uniform Material u_material;
+Material g_material = u_material;
```

Comment:
Yes that's exactly the case the famous:

if( isWindow)
   g_material.specular = vec4(1.,1.,1.,0.5);
else 
   g_material.specular = vec4(0.,0.,0.,0.);
   g_material.diffuse = wallColor;